sonic-xcvrd/xcvrd/dom/dom_mgr.py

```diff
@@ -198,6 +198,47 @@ def post_port_pm_info_to_db(self, logical_port_name, port_mapping, table, stop_e
             else:
                 return xcvrd.SFP_EEPROM_NOT_READY
 
+    def wait_port_initialization(self, delay):
+        logical_port_set = set(self.port_mapping.logical_port_list)
+
+        while logical_port_set:
+            time.sleep(delay)
+
+            for logical_port_name in list(logical_port_set):
+                # Get the asic to which this port belongs
+                asic_index = self.port_mapping.get_asic_id_for_logical_port(logical_port_name)
+                if asic_index is None:
+                   self.log_warning("Got invalid asic index for {}, ignored".format(logical_port_name))
+                   logical_port_set.remove(logical_port_name)
+                   continue
+
+                physical_port_list = self.port_mapping.get_logical_to_physical(logical_port_name)
+                if not physical_port_list:
+                   self.log_warning("Failed to find physical port for lport {}".format(logical_port_name))
+                   logical_port_set.remove(logical_port_name)
+                   continue
+
+                physical_port = physical_port_list[0]
+                if not xcvrd._wrapper_get_presence(physical_port):
```

Comment:
@prgeor Can we instead call `xcvrd_utils.get_transceiver_presence()`?
Also, can we check if `self.task_stopping_event.is_set()`?

Wondering if you want to use `_validate_and_get_physical_port` instead?